SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -2058,3 +2102,49 @@ void CNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container, C
   }
 
 }
+
+void CNSSolver::Setmut_LES(CGeometry *geometry, CSolver **solver_container, CConfig *config) {
+
+  unsigned long iPoint;
+  su2double Grad_Vel[3][3] = {{0.0,0.0,0.0},{0.0,0.0,0.0},{0.0,0.0,0.0}};
+  su2double lenScale, muTurb, rho;
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++){
+
+    /* Get Density */
+    rho = nodes->GetSolution(iPoint, 0);
+
+    /* Velocity Gradients */
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      for (unsigned short jDim = 0 ; jDim < nDim; jDim++)
+        Grad_Vel[iDim][jDim] = nodes->GetGradient_Primitive(iPoint, iDim+1, jDim);
+
+    /* Distance to the wall. */
+    su2double dist = geometry->node[iPoint]->GetWall_Distance(); // Is the distance to the wall used in any SGS calculation?
+
+    /* Length Scale for the SGS model: Cubic root of the volume. */
+    su2double Vol = geometry->node[iPoint]->GetVolume() + geometry->node[iPoint]->GetPeriodicVolume();
+    lenScale = pow(Vol,1./3.);
+
+    /* Compute the eddy viscosity. */
+    if (nDim == 2){
+      muTurb = SGSModel->ComputeEddyViscosity_2D(rho, Grad_Vel[0][0], Grad_Vel[1][0],
+                                                 Grad_Vel[0][1], Grad_Vel[1][1],
+                                                 lenScale, dist);
+    }
+    else{
+      muTurb = SGSModel->ComputeEddyViscosity_3D(rho, Grad_Vel[0][0], Grad_Vel[1][0], Grad_Vel[2][0],
+                                               Grad_Vel[0][1], Grad_Vel[1][1], Grad_Vel[2][1],
+                                               Grad_Vel[0][2], Grad_Vel[1][2], Grad_Vel[2][2],
+                                               lenScale, dist);
+    }
+    /* Set eddy viscosity. */
+    nodes->SetEddyViscosity(iPoint, muTurb);
+  }
+
+  /*--- MPI parallelization ---*/
+
+//  InitiateComms(geometry, config, SGS_MODEL);
```

Comment:
You should be able to remove this. The eddy viscosity is purely local for the SGS, so as long as you have communicated the solution and gradients beforehand, you should be ok, since you are computing the values in the halos as well (nPoint rather than nPointDomain).